Lib/test/test_funcattrs.py

```diff
@@ -478,6 +479,27 @@ def test_builtin__qualname__(self):
         self.assertEqual([1, 2, 3].append.__qualname__, 'list.append')
         self.assertEqual({'foo': 'bar'}.pop.__qualname__, 'dict.pop')
 
+    @support.cpython_only   # See gh-58211
+    def test_builtin__self__(self):
+        import time
+        import builtins
+
+        # builtin function:
+        self.assertEqual(len.__self__, builtins)
+        self.assertEqual(time.sleep.__self__, time)
+
+        # builtin classmethod:
+        self.assertEqual(dict.fromkeys.__self__, dict)
+        self.assertEqual(float.__getformat__.__self__, float)
+
+        # builtin staticmethod:
+        self.assertEqual(str.maketrans.__self__, None)
+        self.assertEqual(bytes.maketrans.__self__, None)
+
+        # builtin bound instance method:
+        self.assertEqual([1, 2, 3].append.__self__, [1, 2, 3])
+        self.assertEqual({'foo': 'bar'}.pop.__self__, {'foo': 'bar'})
+
```

Comment:
Add also a test for the `None` bound method, for example, `None.__repr__.__self__`.

Reply:
What if we test for all built-in constants? I've done that.